indy_common/test/auth/test_role_authorizer.py

```diff
@@ -108,3 +109,38 @@ def test_role_authorizer_is_sig_count_accepted(idr_cache_none_role, req_auth):
 def test_role_authorizer_not_is_sig_count_accepted(idr_cache_none_role, req_auth):
     authorizer = RolesAuthorizer(cache=idr_cache_none_role)
     assert not authorizer.is_sig_count_accepted(req_auth, AuthConstraint(role=TRUSTEE, sig_count=10))
+
+
+def test_role_authorizer_off_ledger_signature_pass(idr_cache, req_auth):
+    authorizer = RolesAuthorizer(cache=idr_cache)
+    req_auth._identifier = 'id_off_ledger'
+    authorized, reason = authorizer.authorize(req_auth, AuthConstraint(role='*', sig_count=1,
+                                                                       off_ledger_signature=True))
+    assert authorized
+
+
+def test_role_authorizer_off_ledger_signature_not_pass(idr_cache, req_auth):
+    authorizer = RolesAuthorizer(cache=idr_cache)
+    req_auth._identifier = 'id_off_ledger'
+    authorized, reason = authorizer.authorize(req_auth, AuthConstraint(role='*', sig_count=1,
+                                                                       off_ledger_signature=False))
+    assert not authorized
+    assert "DID id_off_ledger is not found in the Ledger" in reason
+
+
+def test_role_authorizer_off_ledger_signature_count_2_pass(idr_cache, req_auth):
+    authorizer = RolesAuthorizer(cache=idr_cache)
+    req_auth._identifier = 'id_off_ledger'
+    req_auth.signature = None
+    req_auth.signatures = {'id_off_ledger': 'signature', 'another_id_off_ledger': 'another_signature'}
+    authorized, reason = authorizer.authorize(req_auth, AuthConstraint(role='*', sig_count=2,
+                                                                       off_ledger_signature=True))
+    assert authorized
+
```

Comment:
Please add a test where 1 signature is on ledger, and 1 is off ledger.

Reply:
ok